ArduPlane/mode_takeoff.cpp

```diff
@@ -156,6 +155,10 @@ void ModeTakeoff::update()
         plane.long_failsafe_pending = false;
         plane.failsafe_long_on_event(FAILSAFE_LONG, ModeReason::MODE_TAKEOFF_FAILSAFE);
         }
+    } else { // still climbing to TAKEOFF_ALT; may be loitering
+        plane.calc_throttle();
+        plane.takeoff_calc_roll();
+        plane.takeoff_calc_pitch();
```

Comment:
This whole section would be easier as a nested if I think, removes the duplication. 
```
} else {
        plane.calc_throttle();
        plane.takeoff_calc_roll();
        plane.takeoff_calc_pitch();
        if (altitude >= (alt * 100) - 200) {
            //within 2m of TKOFF_ALT ,or above and loitering
            #if AP_FENCE_ENABLED
                plane.fence.auto_enable_fence_after_takeoff();
            #endif
              // check if in long failsafe, if it is recall long failsafe now to get fs action via events call
              if (plane.long_failsafe_pending) {
                    plane.long_failsafe_pending = false;
                    plane.failsafe_long_on_event(FAILSAFE_LONG, ModeReason::MODE_TAKEOFF_FAILSAFE);
               }
          }
}
```
One problem with moving the `long_failsafe_pending` check is that it is not latching. It will disable if dropping under altitude after having reached it.

Reply:
it should actually be in the last else...thanks...also leaving it there in case someone messes up the takeoff alts configuration ...could put a prearm check for that, but that's creeping elegance IMO

Reply:
OK, can be a nested if with the new logic. 
```
    } else {

        if (altitude >= alt * 100 - 200) { //within 2m of TKOFF_ALT, or above and loitering
#if AP_FENCE_ENABLED
            plane.fence.auto_enable_fence_after_takeoff();
#endif
            plane.calc_nav_roll();
            plane.calc_nav_pitch();
            plane.calc_throttle();
        } else { // still climbing to TAKEOFF_ALT; may be loitering
            plane.calc_throttle();
            plane.takeoff_calc_roll();
            plane.takeoff_calc_pitch();
        }

        //check if in long failsafe, if it is recall long failsafe now to get fs action via events call
        //this should have already been cleared below, but just in case TKOFF_ALT is mis-configured
        if (plane.long_failsafe_pending) {
            plane.long_failsafe_pending = false;
            plane.failsafe_long_on_event(FAILSAFE_LONG, ModeReason::MODE_TAKEOFF_FAILSAFE);
        }
    }
```

Reply:
done